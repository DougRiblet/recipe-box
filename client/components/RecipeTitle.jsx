import React from 'react'

const RecipeTitle = ({ changeActiveRecipe, title }) => (
  <li onClick={changeActiveRecipe}>{title}</li>
)

RecipeTitle.propTypes = {
  changeActiveRecipe: React.PropTypes.func.isRequired,
  title: React.PropTypes.string.isRequired
}

export default RecipeTitle
