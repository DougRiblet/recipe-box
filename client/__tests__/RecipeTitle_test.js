import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme'
import RecipeTitle from '../components/RecipeTitle'

describe('RecipeTitle', () => {
  test('renders correctly', () => {
    const RecipeTitle_tree = renderer.create(
      <RecipeTitle
        key={'0123456'}
        title={'string'}
        changeActiveRecipe={() => changeActiveRecipe('0123456')}
      />
    ).toJSON()
    expect(RecipeTitle_tree).toMatchSnapshot()
  })

  test('renders without crashing', () => {
    const div = document.createElement('div')
    ReactDOM.render(<RecipeTitle
      key={'0123456'}
      title={'string'}
      changeActiveRecipe={() => changeActiveRecipe('0123456')}
    />, div)
  })

  test('should display the passed title', () => {
    const sampleComponent = shallow(<RecipeTitle
      key={'0123456'}
      title={'Kind Veggie Burrito'}
      changeActiveRecipe={() => changeActiveRecipe('0123456')}
    />)
    const displayedLI = <li onClick={changeActiveRecipe}>Kind Veggie Burrito</li>
    expect(sampleComponent.contains(displayedLI)).toEqual(true)
  })
})
