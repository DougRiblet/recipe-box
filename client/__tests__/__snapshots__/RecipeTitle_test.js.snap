// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RecipeTitle Snapshot 1`] = `
<li
  onClick={[Function]}
>
  string
</li>
`;
